analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/PermissiveCors.cs

```diff
@@ -0,0 +1,148 @@
+                          c => ContainsStar((ObjectCreationExpressionSyntax)c.Node, c.SemanticModel));
+
+        private void VisitAttribute(SyntaxNodeAnalysisContext context)
+        {
+            var attribute = (AttributeSyntax)context.Node;
+            if (attribute.IsKnownType(KnownType.System_Web_Http_Cors_EnableCorsAttribute, context.SemanticModel)
+                && IsStar(attribute.ArgumentList.Arguments[0].Expression, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, attribute.GetLocation()));
+            }
+        }
+
+        private static bool IsFirstArgumentAccessControlAllowOrigin(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+            invocation.ArgumentList.Arguments.First().Expression switch
+            {
+                LiteralExpressionSyntax literal => literal.Token.ValueText == AccessControlAllowOriginHeader,
+                MemberAccessExpressionSyntax memberAccess => IsAccessControlAllowOriginProperty(memberAccess, semanticModel),
+                _ => false
+            };
+
+        private static bool IsAccessControlAllowOriginProperty(MemberAccessExpressionSyntax memberAccess, SemanticModel semanticModel) =>
+            memberAccess.Name.Identifier.Text == AccessControlAllowOriginPropertyName
+            && memberAccess.Expression.IsKnownType(KnownType.Microsoft_Net_Http_Headers_HeaderNames, semanticModel);
+
+        private static bool IsSecondArgumentStarString(InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+            IsStar(invocation.ArgumentList.Arguments[1].Expression, semanticModel);
+
+        private static bool IsStar(ExpressionSyntax expressionSyntax, SemanticModel semanticModel) =>
+            expressionSyntax switch
+            {
+                LiteralExpressionSyntax literal => ContainsStar(semanticModel.GetConstantValue(literal)),
+                IdentifierNameSyntax identifier => ContainsStar(semanticModel.GetConstantValue(identifier)),
+                ImplicitArrayCreationExpressionSyntax arrayCreation => ContainsStar(arrayCreation.Initializer.Expressions, semanticModel),
+                ObjectCreationExpressionSyntax objectCreation => ContainsStar(objectCreation, semanticModel),
+                _ => false
+            };
+
+        private static bool ContainsStar(IEnumerable<ExpressionSyntax> expressions, SemanticModel semanticModel) =>
+            expressions.Any(expression => ContainsStar(semanticModel.GetConstantValue(expression)));
+
+        private static bool ContainsStar(Optional<object> constantValue) =>
+            constantValue is {HasValue: true, Value: StarConstant};
+
+        private static bool ContainsStar(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
+            (objectCreation.IsKnownType(KnownType.Microsoft_Extensions_Primitives_StringValues, semanticModel)
```

Comment:
nitpick: wouldn't the argument check be better to be done before? 

do we know if `GetConstantValue` is cheaper than `GetSymbolInfo`? (I would expect it to be cheaper)

Reply:
We now check first for the attribute list to be != null (syntax level), then check the name of the type (syntax level) and only later get the type. I expect that filtering by name will reduce a lot the number of types to get.